src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -95,64 +95,75 @@
          :when (:queryable? projection-value)]
      (keyword projection-key))))
 
+(defn maybe-log-sql [logging-options gen-sql]
+  (if-let [log-id (:log-id logging-options)]
+    (let [sql-output (gen-sql)]
+      (log/debugf "%s:%s:%s" "PDBQuery" log-id (:results-query sql-output))
+      sql-output)
+    (gen-sql)))
+
 (defn query->sql
   "Converts a vector-structured `query` to a corresponding SQL query which will
    return nodes matching the `query`."
-  [query entity version query-options]
-  {:pre  [((some-fn nil? sequential?) query)]
-   :post [(map? %)
-          (jdbc/valid-jdbc-query? (:results-query %))
-          (or (not (:include_total query-options))
-              (jdbc/valid-jdbc-query? (:count-query %)))]}
+  ([query entity version query-options]
+   (query->sql query entity version query-options {}))
+  ([query entity version query-options logging-options]
+   {:pre  [((some-fn nil? sequential?) query)]
+    :post [(map? %)
+           (jdbc/valid-jdbc-query? (:results-query %))
+           (or (not (:include_total query-options))
+               (jdbc/valid-jdbc-query? (:count-query %)))]}
 
-  (cond
-    (= :aggregate-event-counts entity)
-    (aggregate-event-counts/query->sql version query query-options)
+   (maybe-log-sql
+    logging-options
+    #(cond
```

Comment:
Do you think that `(fn []` provides any useful clarity that the anonymous function has no arguments?

Also if wouldn't be clojure PR review without indentation comments, it looks like this is only 1 space indented.

Reply:
👍 I could see changing it to `(fn []` for clarity sake. I wasn't sure about the one space indent at first when my editor kept insisting on it, but then saw [this](https://github.com/bbatsov/clojure-style-guide#arguments-indentation) in the style guide.